TripalCultivate/TripalCultivate-Phenotypes
# G2.10 importer api

**Issue https://github.com/TripalCultivate/TripalCultivate-Phenotypes/issues/10** - Data Importer + validator API

## Motivation
This is the PR sets up base form structure of data importer. It provides 

- logic switch used to move from one stage to the next in the file upload process
- stage indicator that is in sync with current form stage to visually show active stage
- stages are organized into an accordion layout



## What does this PR do?
*Please describe each things this PR does. For example, a PR may 1) solve a specific bug, 2) create an auomated test to ensure it doesn't return.*

1. Create base template of data importer.

## Testing

### Automated Testing
*Please describe each automated test this PR creates and provide a list of the assertions it makes using casual language.*
*Do not just say things like "asserts the array is not empty" but rather say "Ensures that the return value of method X with these parameters is not an empty array".*

ClassName->ImporterShareTest: Test importer base form structure
- Tests custom importer is added as a link (with title and description) in admin/tripal/loaders
- Tests importer is accessible with correct user permission (allow tripal import)
- Tests that page title matches expected form title.
- Tests stage indicators show the correct stage
- Test that accordion exposes/expands the correct stage

### Manual Testing
*Describe in detail how someone should manually test this functionality.*
*Make sure to include whether they need to build a docker from scratch, create any records, etc.*

1. Switch to branch Importer-api of trpcultivate_phenotypes
2. Install and enable the module
3. Access Tripal page that displays all implementation of Importer plugin in admin/tripal/loaders
4. Click on Tripal Cultivate: Open Science Phenotypic Data link
5. Navigate each stage using the Next Page button in each stage in the accordion layout

### Advanced Options
To keep form simple and code base light, I will leave the advanced options as is, no theme, no JS behaviour nor any modifications.

![Screen Shot 2023-07-19 at 3 02 58 PM](https://github.com/TripalCultivate/TripalCultivate-Phenotypes/assets/15472253/8e0ae411-a821-4dd0-82b7-5e1745875073)

Note to self: Reminder about the field element in the help section to say -
Alter the value of schema only when your administrator instructed to do so
